apps/src/sites/code.org/pages/public/learn/local.js

```diff
@@ -117,31 +127,189 @@ function loadMap(locations) {
+      type: 'Feature',
+      properties: {
+        description: location.html,
+        title: location.title,
+        index
+      },
+      geometry: {
+        type: 'Point',
+        coordinates: [parseFloat(location.lon), parseFloat(location.lat)]
+      }
+    }));
+    gmap = new mapboxgl.Map({
+      container: 'gmap',
+      style: 'mapbox://styles/mapbox/streets-v11',
+      zoom: 10,
+      minZoom: 1,
+      center: [lng, lat]
+    });
+    gmap.on('load', function() {
+      gmap.loadImage(
+        'https://docs.mapbox.com/mapbox-gl-js/assets/custom_marker.png',
+        function(error, image) {
+          // do stuff
+          if (error) {
+            throw error;
+          }
+          gmap.addImage('custom-marker', image);
+          gmap.addSource('places', {
+            type: 'geojson',
+            data: {
+              type: 'FeatureCollection',
+              features: featureList
+            }
+          });
+          // Add a layer showing the places.
+          gmap.addLayer({
+            id: 'places',
+            type: 'symbol',
+            source: 'places',
+            layout: {
+              'icon-image': 'custom-marker',
+              'icon-allow-overlap': true
+            }
+          });
 
-  if (locations.length > 0) {
-    map_options.force_generate_controls = true;
-    map_options.locations = locations;
-    map_options.afterOpenInfowindow = function(index, location, marker) {
-      setDetailsTrigger(index, location, marker);
+          // When a click event occurs on a feature in the places layer, open a popup at the
+          // location of the feature, with description HTML from its properties.
+          gmap.on('click', 'places', function(e) {
+            var coordinates = e.features[0].geometry.coordinates.slice();
+
+            // Ensure that if the map is zoomed out such that multiple
+            // copies of the feature are visible, the popup appears
+            // over the copy being pointed to.
+            while (Math.abs(e.lngLat.lng - coordinates[0]) > 180) {
+              coordinates[0] += e.lngLat.lng > coordinates[0] ? 360 : -360;
+            }
+            var activeItem = document.querySelector('li.active');
+            if (activeItem) {
+              activeItem.classList.remove('active');
+            }
+            document
+              .querySelector(`#control-${e.features[0].properties.index}`)
+              .classList.add('active');
+            createPopUp(e.features[0]);
+          });
+
+          // Change the cursor to a pointer when the mouse is over the places layer.
+          gmap.on('mouseenter', 'places', function() {
+            gmap.getCanvas().style.cursor = 'pointer';
+          });
+
+          // Change it back to a pointer when it leaves.
+          gmap.on('mouseleave', 'places', function() {
+            gmap.getCanvas().style.cursor = '';
+          });
+        }
+      );
+    });
+    if (featureList.length > 0) {
```

Comment:
What is this code doing?

Reply:
This makes the side menu with the list of schools. Google maps does it for us right now but for mapbox you have to build it yourself (https://docs.mapbox.com/help/tutorials/building-a-store-locator/)

Reply:
Ahhh makes sense thanks! Do you mind adding a quick comment saying that?

Reply:
sure!